### PR TITLE
Proposition : model number handling

### LINKED ISSUE #4295: Companion 2.2-RC9 Saving in .otx removes unused models and renumbers following models
Companion 2.2-RC9 Radio X9D
I don't know if this is a bug or "working as designed" but if there are any unused models (gaps), saving in .otx format removes the unused models and moves all models after that up to fill those slots (changing the model numbers accordingly).  Saving in EEPE maintains the model numbers as they are on the transmitter as it always has.
capture

### PR DESCRIPTION
**Pull request by 3djc**
This is a proposal to fix #4295

### CODE REVIEW

**Comment by projectkk2glider** [comment]
Your current proposal is too hard-coded for my liking (support max 100 models)

I have this proposal:
 * if Horus - no change
 * if not Horus - use this format:
 `<position><delimiter><model name>`
 
Where:
`<position>` is a decimal number
`<delimiter>` one or more whitespace (space, tab) characters (use one space for writing, accept one or more when reading)
`<model name>` normal model name

Use `QStringList QString::split(const QRegExp &rx, SplitBehavior behavior = KeepEmptyParts) const` to split each line into two parts and convert first part to number.

**Comment by 3djc** [comment]
Indeed much better. There are a fex chars  we can use as delimiters, as allowed chars for model names is quite limited, thats why I used  ^ that is not allowed for model names

File: companion/src/storage/categorized.cpp [hunk 1]
```diff
@@ -54,10 +56,20 @@ bool CategorizedStorageFormat::load(RadioData & radioData)
...
+        QByteArray curline;
+        if (line.contains(FIELD_DELIMITER)) {
+          modelIndex=line.mid(0,line.indexOf(FIELD_DELIMITER)).toInt();
+          curline = line.mid(line.indexOf(FIELD_DELIMITER)+1);
+          qDebug() << "Detected" << curline;
+        }
+        else {
+          curline = line.data();
+        }
+
```
Comment (schwabe) [inline companion/src/storage/categorized.cpp]: Qstring curline = QString(curline).trimmed()  to remove all the potential extra white whitespaces

**Comment by bsongis** [comment]
Excellent!

**Pull request merged**